h2c/server.go

```diff
@@ -44,12 +52,46 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	s.trackConn(conn, true)
+	defer s.trackConn(conn, false)
 
 	new(http2.Server).ServeConn(bufConn{conn, bufrw}, &http2.ServeConnOpts{
 		Handler: s.HTTP2Handler,
 	})
 }
 
+func (s *Server) Shutdown(ctx context.Context) error {
```

Comment:
This is poorly named, perhaps. It does not in any way actually shutdown the server, it assumes that the http 1.1 server has already been shutdown, and simply waits for all hijacked connections to finish.

Reply:
I wonder if we should consider blocking new connections as part of this, like returning a 503. Right now `Shutdown` is basically "wait for there to be no connections", which on even a moderately busy server could be forever.

Reply:
The intention is that the caller has to shutdown the active listener (which we don't have access to) by calling `Shutdown` on the actual `http` server first. I've added a comment tot clarify that.

I don't really have strong opinions on returning a 503. Though it shouldn't be necessary if the caller closes the listener first (as intended)